### DISCUSSION TITLE
Editing settings.json file - How do you get around the permissions?

### DISCUSSION THREAD

**Discussion by maddog0487** [nebhead/PiTender#17] [Q&A] [answered]
Hello Nebhead, I started the Bartender project a few months back and got hung up trying to get the older code updated to Python 3, I stumbled across PiTender and think it is great.  I am fairly new to the world of Pi and can't figure out how to get around the file permissions to modify the code to add more pumps; I am assuming this is because the code is running but cannot figure out how to stop the execution to modify it.  I used the auto installer which worked great.  Any suggestions would be appreciated.  I will upload some pictures after my next phase of construction.

**Comment by nebhead** [comment]
Hey @maddog0487 - It could be that the code is running, or it may just be because you need to use `sudo` to gain superuser permissions.

To stop the code from running, you should be able to go into the supervisor WebUI assuming you opted-in to this when you ran the installation script.  From your browser, navigate to the same IP address you use to get the the normal WebUI, but with :9001.  For example:
![image](https://user-images.githubusercontent.com/19876886/235404335-cd4bb89a-2102-4dc4-957e-3031708683f9.png)

Type in your username/password (I personally just use user:pass).  
![image](https://user-images.githubusercontent.com/19876886/235404449-4867fa74-e2da-4771-82b7-cfbc34ee7cbb.png)

Then from the user interface, select stop all (this will look different because this is a different project):
![image](https://user-images.githubusercontent.com/19876886/235404492-9488fd43-66e8-4f19-a16a-3f6406f7e5e3.png)

Then, you can SSH into your pi as usual, navigate to /usr/local/bin/pitender, and then edit the files that you want to edit using `sudo` for superuser permissions.  For example:

```bash
$ cd /usr/local/bin/pitender 
$ sudo nano app.py 
$ sudo nano settings.json 
```

Hope this helps!

**Reply by maddog0487** [reply]
Thanks for the quick reply!  I will give it a try tonight and post some pictures of my build.  Thanks again for this code.

**Reply by maddog0487** [reply, marked as answer]
Ok, tried multiple things, ended up loading an old version of Raspberry Pi OS Lite (Legacy), I then followed your guidance and it is now working.  Now I just need to troubleshoot my pumps!  Bought them a while back and the first time using one of them is tripping the overload on my power supply.
